pkg/scheduler/api/resource_info.go

```diff
@@ -224,40 +224,46 @@ func (r *Resource) Multi(ratio float64) *Resource {
 
 // Less checks whether a resource is less than other
 func (r *Resource) Less(rr *Resource) bool {
-	lessFunc := func(l, r float64) bool {
-		return l < r
-	}
+	var isLess bool
 
-	if !lessFunc(r.MilliCPU, rr.MilliCPU) {
+	if r.MilliCPU > rr.MilliCPU {
 		return false
 	}
-	if !lessFunc(r.Memory, rr.Memory) {
+	if r.MilliCPU < rr.MilliCPU {
+		isLess = true
+	}
+
+	if r.Memory > rr.Memory {
 		return false
 	}
+	if r.Memory < rr.Memory {
+		isLess = true
+	}
 
 	if r.ScalarResources == nil {
 		if rr.ScalarResources != nil {
-			for _, rrQuant := range rr.ScalarResources {
-				if rrQuant <= minMilliScalarResources {
+			for _, rQuant := range rr.ScalarResources {
+				if rQuant <= minMilliScalarResources {
 					return false
 				}
+				isLess = true
 			}
 		}
-		return true
 	}
 
-	if rr.ScalarResources == nil {
-		return false
-	}
-
-	for rName, rQuant := range r.ScalarResources {
-		rrQuant := rr.ScalarResources[rName]
-		if !lessFunc(rQuant, rrQuant) {
-			return false
+	if rr.ScalarResources != nil {
+		for name, lQuant := range r.ScalarResources {
+			rQuant := rr.ScalarResources[name]
+			if lQuant > rQuant {
+				return false
+			}
+			if lQuant < rQuant {
+				isLess = true
+			}
 		}
 	}
 
-	return true
+	return isLess
 }
 
 // LessEqualStrict checks whether a resource is less or equal than other
```

Comment:
l think for the issue #1036, maybe function `LessEqualStrict ` and `LessEqual ` can be changed together)